### ISSUE TITLE
Make ReactComponent capable of supporting multiple Panel versions

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#7213]
I've been hit several times by the fact that the Panel `ReactComponent` uses a specific version of `React`. Currently 18.3.1.

- The specific version is not publicly documented.
- It's not explained that you will have to use React library versions compatible with the exact version of React Panel is providing
- If you import versions of React libraries that are not compatible with Panels version of React you get hard to understand error messages and need guidance from Philipp that versioning is the problem.
- There is a high risk that custom `ReactComponent` will only work for a narrow band of Panel versions because Panel will and should keep the react version updated. This makes these components hard to maintain.

Please remove the issues above.

**Comment by philippjfr** [comment]
The react version can be configured by declaring `_react_version`.

**Comment by MarcSkovMadsen** [comment]
It is not clear to me if I can distribute a package using a React version not the same as the one the current version of Panel uses.

**Comment by philippjfr** [comment]
>It is not clear to me if I can distribute a package using a React version not the same as the one the current version of Panel uses. And not the same version as some other ReactComponent extension version.

As I mentioned above, there is no such thing as the "version of React that Panel uses", the version is configurable on the component itself.

**Comment by MarcSkovMadsen** [comment]
Ok. Then we should document it. Because if extension developers do not know this their extension packages will only work for very specific versions of Panel.

**Comment by philippjfr** [comment]
Absolutely, it is documented as part of the compile and bundle how to guide now but should be made more prominent in the reference guide as well.
